daemon/daemon.go

```diff
@@ -511,6 +530,20 @@ var (
 	rebootMaxTentatives    = 3
 )
 
+func (d *Daemon) writeMaintenanceFile(rst state.RestartType) error {
+	b, err := json.Marshal(maintenanceForRestartType(rst))
```

Comment:
we also discussed storing the boot-id in it, is that not needed?

Reply:
no that ended up not being needed since we distinguish from a snapd restart and a system reboot in the json anyways